js/src/setup-ads/ads-stepper/index.js

```diff
@@ -27,11 +30,45 @@ import {
  */
 const AdsStepper = ( { formProps } ) => {
 	const [ step, setStep ] = useState( '1' );
+	const initHasAdsConnectionRef = useRef( null );
+	const { hasFinishedResolution: hasResolvedGoogleAccount } =
+		useGoogleAccount();
+
+	const {
+		googleAdsAccount,
+		hasFinishedResolution: hasResolvedGoogleAdsAccount,
+		hasGoogleAdsConnection,
+	} = useGoogleAdsAccount();
+
+	const {
+		hasAccess,
+		hasFinishedResolution: hasResolvedAdsAccountStatus,
+		step: adsAccountSetupStep,
+	} = useGoogleAdsAccountStatus();
+
 	useEventPropertiesFilter( FILTER_ONBOARDING, {
 		context: CONTEXT_ADS_ONBOARDING,
 		step,
 	} );
 
+	if (
+		! hasResolvedGoogleAccount ||
```

Comment:
Is it necessary to check this condition?

Reply:
Fair point, yes we don't need this check. Other check's indirectly confirms this.